init.lua

```diff
@@ -240,18 +239,19 @@ local function smooth(pos1, pos2, deadzone)
 				local y = old_height-1
 				while y >= new_height do
 					local index = index_z + (offset.y + y) * stride.y
-					if data[index] == c_dirt then data[index] = c_air end
+					if data[index] ~= c_air then data[index] = c_air end
 
 					count = count + 1
 					y = y - 1
 				end
 			elseif old_height < new_height then
 				-- need to add nodes
 				local y = old_height
+				local c_target = minetest.get_node(vector.new(index_x, old_height, index_z))
```

Comment:
no this won't work at all, you need to read the node ID from the `data` array instead

Reply:
Ahhh I see. I'll correct it shortly, as I'm not at my PC right now.

I'll also test it too :P